engine/apps/schedules/models/custom_on_call_shift.py

```diff
@@ -301,13 +301,16 @@ def convert_to_ical(self, time_zone="UTC"):
                     if not start:  # means that rotation ends before next event starts
                         all_rotation_checked = True
                         break
-                    elif start >= self.rotation_start:  # event has already started, generate iCal for each user
+                    elif (
+                        self.source == CustomOnCallShift.SOURCE_WEB and start + self.duration > self.rotation_start
```

Comment:
To be sure, do we need to check for `SOURCE_WEB` here? wouldn't all other shifts have `start >= rotation_start`? (assuming all other shifts got `rotation_start` set to `shift_start`)

Reply:
We don't set `rotation_start` to `shift_start` because in this case we will break repetition rules for future events. We keep in iCal theirs original start date and recalculate theirs start with respect to `rotation_start` in other place. And only for `SOURCE_WEB`, we assume that the event can start after the original start, that's why we're checking event type here